### PR TITLE
Fix Dropdown's `defaultValue`/`value` props behaviour

### PR DESCRIPTION
**Pull request by mickaelzhang**
## Screenshot / video

This message is showing up when `value` is provided
Screenshot 2022-09-21 at 15 45 34

### CODE REVIEW

File: src/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -63,6 +64,10 @@ export const Dropdown = forwardRef(function Dropdown(
   }: DropdownProps,
   ref: ForwardedRef<HTMLSelectElement>,
 ) {
+  const [value, setValue] = useControllableState({
+    initialState: defaultValue,
+    stateProp: valueProp,
+  })
```
Comment (mickaelzhang) [inline src/Dropdown/Dropdown.tsx:70]: `Dropdown` behaviour when used is still the same
Internally tho, it's always controlled even when only `defaultValue` is provided

File: src/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -94,17 +99,13 @@ export const Dropdown = forwardRef(function Dropdown(
...
-          }
           disabled={disabled || list.length < 1}
-          onChange={(e: FormEvent<HTMLSelectElement>) => {
-            onSelect && onSelect(e.currentTarget.value)
-            onInputChange && onInputChange(e)
+          onChange={(event) => {
+            const value = event.currentTarget.value
+
+            onSelect?.(value)
+            onInputChange?.(event)
```
Comment (mickaelzhang) [inline src/Dropdown/Dropdown.tsx:107]: Not sure a second event is needed tbh, is it used?

File: src/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -198,23 +199,16 @@ const StyledSelect = styled.select<UsesOutline>`
...
-  &:focus-within {
-    border: 2px solid ${theme.colors.outline};
-  }
-`}
-  ${({ value }) =>
-    value &&
-    value != '' &&
-    `
-    border: 2px solid ${theme.colors.outline};
-`}
```
Comment (mickaelzhang) [inline src/Dropdown/Dropdown.tsx:216]: I've removed this, I think it was not intended, there is a border when the component is controlled

File: src/Dropdown/Dropdown.tsx (hunk 2 above)
Comment (fleonard) [reply src/Dropdown/Dropdown.tsx:107]: yeah I think FNOL added it cause `onSelect` return the string value while they needed the actual event

**Review by fleonard — APPROVED** [review]

File: src/Dropdown/Dropdown.tsx (hunk 2 above)
Comment (josianed) [reply src/Dropdown/Dropdown.tsx:107]: `onInputChange` is not used at all throughout FNOL. It was added before the correct use of the input components & react-hook-forms was established and can be removed now

**Pull request merged**